python_modules/airline-demo/airline_demo/pipelines.py

```diff
@@ -109,106 +112,212 @@ def _create_postgres_engine(username, password, hostname, db_name):
+#             table_name,
+#         )
+#     elif db_dialect == 'postgres':
+#         data_frame.write \
+#         .option('driver', 'org.postgresql.Driver') \
+#         .mode('overwrite') \
+#         .jdbc(
+#             resources.db_url,
+#             table_name,
+#         )
+#     else:
+#         raise NotImplementedError(
+#             'No implementation for db_dialect "{db_dialect}"'.format(db_dialect=db_dialect)
+#         )
+
+def postgres_table_loader(data_frame, table_name, resources):
+    data_frame.write \
+    .option('driver', 'org.postgresql.Driver') \
+    .mode('overwrite') \
+    .jdbc(
+        resources.db_info.url,
+        table_name,
         )
-    elif db_dialect == 'postgres':
+
+def redshift_table_loader(data_frame, table_name, resources):
+    data_frame.write \
+    .format('com.databricks.spark.redshift') \
+    .option('tempdir', resources.redshift_s3_temp_dir) \
+    .mode('overwrite') \
+    .jdbc(
+        resources.db_info,
+        table_name,
+    )
+
+
+LocalSparkResourceDefinition = ResourceDefinition(
+    resource_fn=lambda _info: _create_spark_session_local(),
+    config_field=None,
+)
+
+S3ResourceDefinition = ResourceDefinition(
+    resource_fn=lambda _info: _create_s3_session(),
+    config_field=None,
+)
+
+
+RedshiftConfigDict = types.ConfigDictionary(
+    'RedshiftConfigDict',
+    {
+        'redshift_username': Field(types.String),
+        'redshift_password': Field(types.String),
+        'redshift_hostname': Field(types.String),
+        'redshift_db_name': Field(types.String),
+        'redshift_s3_temp_dir': Field(types.String),
+    },
+)
+
+# RedshiftDbUrlResource = ResourceDefinition(
+#     resource_fn = lambda info: _create_redshift_db_url(**info.config),
+#     config_field=types.Field(RedshiftConfigDict),
+# )
+
+
+# RedshiftEngineResource = ResourceDefinition(
+#     resource_fn = lambda info: _create_redshift_engine(**info.config),
+#     config_field=types.Field(RedshiftConfigDict),
+# )
+
+RedshiftDbInfoResource = ResourceDefinition(
+    resource_fn=lambda info: RedshiftDbInfo(
+        _create_postgres_engine(
+            info.config['redshift_username'],
+            info.config['redshift_password'],
+            info.config['redshift_hostname'],
+            info.config['redshift_db_name'],
+        ),
+        _create_redshift_db_url(
+            info.config['redshift_username'],
+            info.config['redshift_password'],
+            info.config['redshift_hostname'],
+            info.config['redshift_db_name'],
+        ),
+        info.config['redshift_s3_temp_dir'],
+    ),
+    config_field=types.Field(RedshiftConfigDict),
+)
+
+PostgresConfigDict = types.ConfigDictionary(
+    'PostgrestConfigDict',
+    {
+        'postgres_username': Field(types.String),
+        'postgres_password': Field(types.String),
+        'postgres_hostname': Field(types.String),
+        'postgres_db_name': Field(types.String),
+    },
+)
+
+class DbInfo:
```

Comment:
cute, this makes it clear to me how users can be pointed to extend library resource classes